packages/react-instantsearch-hooks-dom/src/widgets/CurrentRefinements.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { useCurrentRefinements } from 'react-instantsearch-hooks';
+
+import { CurrentRefinements as CurrentRefinementsUiComponent } from '../ui/CurrentRefinements';
+
+import { isModifierClick } from './lib/isModifierClick';
+
+import type { CurrentRefinementsProps as CurrentRefinementsUiComponentProps } from '../ui/CurrentRefinements';
+import type { UseCurrentRefinementsProps } from 'react-instantsearch-hooks';
+
+export type CurrentRefinementsProps = Omit<
+  CurrentRefinementsUiComponentProps,
+  'disabled' | 'onClick' | 'items' | 'hasRefinements'
+> &
+  UseCurrentRefinementsProps;
+
+export function CurrentRefinements(props: CurrentRefinementsProps) {
+  const {
+    includedAttributes,
+    excludedAttributes,
+    transformItems,
+    ...otherProps
+  } = props;
+  const { items, refine, canRefine } = useCurrentRefinements(
+    {
+      includedAttributes,
+      excludedAttributes,
+      transformItems,
+    },
+    {
+      $$widgetType: 'ais.currentRefinements',
+    }
+  );
+
+  return (
+    <CurrentRefinementsUiComponent
+      {...otherProps}
+      hasRefinements={canRefine}
+      items={items.map((item) => ({
+        label: item.label,
+        category: item.refinements.map((refinement) => ({
+          label: refinement.label,
+          onClick: (event) => {
+            if (isModifierClick(event)) {
+              return;
+            }
+            event.preventDefault();
+
+            refine(refinement);
```

Comment:
this onClick should be part of the ui component IMO and category passing refine as expected. It's similarly coupled in both cases, but this mapping doesn't seem needed to me

Reply:
I find the coupling looser this way, and less leaky of the InstantSearch API. We do the same in [`SearchBox`](https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-hooks-dom/src/widgets/SearchBox.tsx#L58-L59)—we're not passing `setValue` directly, but callbacks instead.

Open to discuss this (along with @dhayab and @francoischalifour as well), as this will impact how we design the next UI components.

Reply:
I see why it could make sense, but to me an onClick is part of the ui component, not something that "framework wrappers" should care about. The contract of `item.refinements` is already defined by the connector, and I don't really see why we should stray

Reply:
My take is that this should be in the UI component because it should handle accessibility for us (we don't want to write this code in all the flavors, we want consistency):

```js
if (isModifierClick(event)) {
  return;
}
event.preventDefault();
```

And this should be in a `onToggle()` callback, because _this_ is specific to InstantSearch:

```js
refine(refinement);
```

That's how I've designed the other components.

Reply:
I agree with moving the click modifying part in the UI component—I initially hesitated to do so. We indeed don't want to replicate this, ideally there should be a single `isModifierClick` function, not one per flavor.

And indeed we probably don't need to map `onClick` callbacks to have one per category. We can have a top-level `onClick` or `onRemove` callback within which we call `refine`. The only important thing for me is that we shouldn't have a `refine` prop on the UI component because that's an InstantSearch API, not a UI concern.

Reply:
I don't think we can create that api easily, or will need to map back to the item, `refine` isn't global for current refinements I thought

Reply:
@Haroenv I think it is. There's a [single `refine` function](https://www.algolia.com/doc/api-reference/widgets/current-refinements/react-hooks/#widget-param-refine) returned by the connector, and it takes a refinement. So there can be a single callback.

I'm good with both local `onClick` functions on each items (current situation) or a single `onRemove` callback at the top level (@dhayab's proposal), there shouldn't be any differences.

Reply:
I'm ok with a global onRemove and items={items} :)

Reply:
@Haroenv @dhayab @francoischalifour Updated in 2e6607bc. There's slightly more coupling now since `onRemove` exposes a `CurrentRefinementsConnectorParamsRefinement`, but we no longer map `items`.

If we want to decouple later on, we can map and have local `onClick` functions on items (my personal preference). I think this is okay, even after we move components to `ui-components`. We can have breaking changes in `ui-components` without having breaking changes in InstantSearch.